src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWING_COEFFICIENT = 1.5d;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayTooSmall();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+        } else {
+            checkArrayIndex(index);
+            growIfArrayTooSmall();
+            System.arraycopy(array, index, array,index + 1,size - index);
```

Comment:
Don't forget about code formatting. [Reference](https://www.jetbrains.com/idea/guide/tips/reformat-file/#:~:text=You%20can%20use%20%E2%8C%98%E2%8C%A5,L%20(Windows%2FLinux).).

```suggestion
            System.arraycopy(array, index, array, index + 1, size - index);
```